openedx/adg/common/lib/mandrill_client/client.py

```diff
@@ -39,7 +39,6 @@ class MandrillClient(object):
     def __init__(self):
         self.mandrill_client = mandrill.Mandrill(settings.MANDRILL_API_KEY)
 
-    @mandrill_exception_handler
```

Comment:
Why are we removing this decorator and instead re-writing code for exception handling?

Reply:
In case of cancelling scheduled emails, we don't want to raise the exception as we are doing in the decorator. The case is that we do not know if a scheduled email is sent and if we try to cancel it then the decorator will raise an exception and the task will break.

Reply:
This scenario too can be handled by using [parameterized decorator/decorator](https://www.kite.com/python/answers/how-to-pass-parameters-to-a-decorator-in-python) factory. You can pass a parameter to the decorator `raise = True/False`.